### PR TITLE
Add mcxc444 base enablement

### PR DESCRIPTION
**Pull request by NeilChen93**
Add basic enablement of MCXC444 soc, frdm_mcxc444 board

### CODE REVIEW

File: boards/nxp/frdm_mcxc444/doc/frdm_mcxc444.png
Comment (danieldegrasse) [inline boards/nxp/frdm_mcxc444/doc/frdm_mcxc444.png:1]: Please convert the png into a `webp` format, to save space

Comment (NeilChen93) [reply boards/nxp/frdm_mcxc444/doc/frdm_mcxc444.png:1]: Updated.

File: boards/nxp/frdm_mcxc444/doc/frdm_mcxc444.webp
Comment (nordicjm) [inline boards/nxp/frdm_mcxc444/doc/frdm_mcxc444.webp:1]: Remove spacing around outside of image, then put it through https://tinypng.com/

File: boards/nxp/frdm_mcxc444/doc/index.rst [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+- `MCX C44x Datasheet`_
+- `MCX C44x Reference Manual`_
+- `FRDM-MCXC444 Website`_
+- `FRDM-MCXC444 User Manual`_
+- `FRDM-MCXC444 Design Files`_
+
+Supported Features
+==================
+
+The FRDM-MCXC444 board configuration supports the following hardware features:
```
Comment (nordicjm) [inline boards/nxp/frdm_mcxc444/doc/index.rst:48]: see other PRs

**Review by nordicjm — CHANGES_REQUESTED** [review]

**Review by nordicjm — DISMISSED** [review]

**Review by danieldegrasse — DISMISSED** [review]

**Review by mmahadevan108 — DISMISSED** [review]

**Comment by mmahadevan108** [comment]
@NeilChen93, can you take a look at the Documentation Build warnings.

**Review by dleach02 — APPROVED** [review]

**Review by EmilioCBen — APPROVED** [review]

**Review by mmahadevan108 — APPROVED** [review]

**Pull request merged**